src/Common/src/TypeSystem/Common/TargetDetails.cs

```diff
@@ -33,6 +33,22 @@ public enum TargetOS
     }
 
     /// <summary>
+    /// Specifies 
+    /// </summary>
+    public enum TargetApplicationModel
+    {
+        Unknown,
+        /// <summary>
+        /// Cross-platform console model
+        /// </summary>
+        CoreRT,
+        /// <summary>
+        /// Windows-specific UWP model
+        /// </summary>
+        NetNative
```

Comment:
ProjectN is not the public name for that though; it's .NET Native.  Have we standardized on calling it ProjectN in code?